go/inst/instance.go

```diff
@@ -271,7 +304,10 @@ func (this *Instance) IsReplica() bool {
 
 // IsMaster makes simple heuristics to decide whether this instance is a master (not replicating from any other server)
 func (this *Instance) IsMaster() bool {
-	return !this.IsReplica()
+	return !this.IsReplica() &&
+		((this.IsReplicationGroupMember() &&
+			this.ReplicationGroupMemberRole != GroupReplicationMemberRoleSecondary) ||
+			!this.IsReplicationGroupMember())
```

Comment:
too much `&& || (( ))` complexity. I can hardly parse.
```suggestion
        if this.IsReplica() {
          return false
        }
	if !this.IsReplicationGroupMember() {
	  return false
	}
	if this.IsReplicationGroupMember() && this.ReplicationGroupMemberRole == GroupReplicationMemberRoleSecondary {
	  return false
	}
	return true
```

Reply:
Fixed